File: src/java/_1685_Sum_of_Absolute_Differences_in_a_Sorted_Array.java
class Solution {
    public int[] getSumAbsoluteDifferences(int[] nums) {
        int prefix = 0, suffix = 0, n = nums.length;
        int[] ans = new int[n];

        for(int num : nums)
            suffix += num;

        for(int i = 0; i < n; i++){
            int num = nums[i];
            ans[i] = num * i - prefix + suffix - num * (n - i);
            prefix += num;
            suffix -= num;
        }

        return ans;
    }
}

File: src/python/_1685_Sum_of_Absolute_Differences_in_a_Sorted_Array.py
class Solution:
    def getSumAbsoluteDifferences(self, nums: List[int]) -> List[int]:
        n = len(nums)
        prefix, suffix = 0, sum(nums)
        ans = []

        for i, num in enumerate(nums):
            ans.append(num * i - prefix + suffix - num * (n - i))
            prefix += num
            suffix -= num

        return ans
